fastapi_jsonapi/data_layers/filtering/sqlalchemy.py

```diff
@@ -1,290 +1,353 @@
 """Helper to create sqlalchemy filters according to filter querystring parameter"""
-from typing import Any, List, Tuple, Type, Union
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Set,
+    Type,
+    Union,
+)
 
 from pydantic import BaseModel
 from pydantic.fields import ModelField
 from sqlalchemy import and_, not_, or_
-from sqlalchemy.orm import InstrumentedAttribute, aliased
-from sqlalchemy.sql.elements import BinaryExpression
+from sqlalchemy.orm import aliased
+from sqlalchemy.orm.attributes import InstrumentedAttribute
+from sqlalchemy.orm.util import AliasedClass
+from sqlalchemy.sql.elements import BinaryExpression, BooleanClauseList
 
-from fastapi_jsonapi.data_layers.shared import create_filters_or_sorts
 from fastapi_jsonapi.data_typing import TypeModel, TypeSchema
 from fastapi_jsonapi.exceptions import InvalidFilters, InvalidType
 from fastapi_jsonapi.schema import get_model_field, get_relationships
-from fastapi_jsonapi.splitter import SPLIT_REL
-from fastapi_jsonapi.utils.sqla import get_related_model_cls
 
-Filter = BinaryExpression
-Join = List[Any]
+RELATIONSHIP_SPLITTER = "."
 
-FilterAndJoins = Tuple[
-    Filter,
-    List[Join],
-]
+RelationshipPath = str
```

Comment:
может сделаем `RelationshipPath: TypeAlias = str`?

Reply:
Упали тесты на python 3.9 
`ImportError: cannot import name 'TypeAlias' from 'typing'`

Reply:
ну и ладно

> Deprecated since version 3.12: [TypeAlias](https://docs.python.org/3/library/typing.html#typing.TypeAlias) is deprecated in favor of the [type](https://docs.python.org/3/reference/simple_stmts.html#type) statement, which creates instances of [TypeAliasType](https://docs.python.org/3/library/typing.html#typing.TypeAliasType)